scriptmodules/emulators/jzintv.sh

```diff
@@ -61,17 +70,60 @@ function install_jzintv() {
 }
 
 function configure_jzintv() {
-    mkRomDir "intellivision"
+    local system="intellivision"
+    mkRomDir $system
 
-    local options=(
-        --displaysize="%XRES%x%YRES%"
-        --quiet
-        --rom-path="$biosdir"
-        --voice=1
-    )
+    local start_script="$romdir/$system/+Start jzintv.sh"
+    cat > "$start_script" << _EOF_
+#! /usr/bin/env bash
+
+# \$1: width of display
+# \$2: height of display
+# \$3: --ecs=1, optional
+# \$4,5,6...: more optional parameters
+# last parameter: %ROM%
+
+jzintv_bin="$md_inst/bin/jzintv"
+
+disp_w=\$1; shift
+disp_h=\$1; shift
+
+min=\$(( \$disp_w < \$disp_h ? \$disp_w : \$disp_h ))
+if [[ \$min = \$disp_h ]] ; then
+    intv_w=\$(echo \$min | awk '{ printf "%0.f\n", \$0 * 1.3333 + .5 }')
+    intv_h=\$min
+else
+    # 90/270 rotated displays
+    intv_w=\$min
+    intv_h=\$(echo \$min | awk '{ printf "%0.f\n", \$0 / 1.3333 }')
+fi
+
+# set --gfx-verbose instead of --quiet for verbose output
+options=(
+    -f1 # fullscreen
+    --quiet
+#    --gfx-verbose
+    --displaysize="\${intv_w}x\${intv_h}"
+    --rom-path="$biosdir"
+    --voice=1
+)
+
+# ingest additional options
+while [[ \$# -gt 1 ]] ; do
+    options+=(\$1); shift
+done
+
+rom=\$1
+echo "Launching: \$jzintv_bin \${options[*]} \"\$rom\"" >> /dev/shm/runcommand.log
+
+pushd "$romdir/$system" > /dev/null
+\$jzintv_bin \${options[*]} "\$rom"
+popd
+_EOF_
```

Comment:
Can you explain better what is the launcher wrapper supposed to do ? I'm guessing one function would be to set the display size to a 4:3 video mode, but what does the rotation has to do with it ?

Reply:
Right, one is to limit the emulator display to 4:3 on a physical display with different aspect ratio.

The rotation is an educated guess. On 90/270 degree I assume the XRES and YRES are flipped (ie. 1080 and 1920). If not, what options do exsist to detect 90/270 mounted displays?

Reply:
If the display is flipped, then wouldn't the emulator display also be flipped ? I don't see how this would be useful unless the emulator has a rotation feature (like ES or RetroArch).
For detecting if the display is rotated, I don't think that's possible outside X11 (using `xrandr`). For a RPI-03 with `dispmanx`, the display is rotated by the firmware and it's transparent to the system, for KMSDRM there's no global rotation configuration that works, each DRM client should handle the orientation hint (DSM prop) accordingly.
Leaving aside the rotation, which would bring something new, wouldn't it be better if the user decided (via `runcommand`'s video mode selection) the aspect ration ? Your patch would handle the letterboxing when the user selects a 4:3 aspect ration and we can pass %XRES% and %YRES% directly to `jzintv`, without the need for a wrapper script ?

Reply:
OK, that looks cleaner, but I think it would be cleaner to replace `bc` and `awk` calls with `python` (which is a system package and always available).
* The `bc` calculation can be replaced with:
   ``` shell
   local do_pillarboxing="$(python -c 'print(800/600>=4/3)')"
   if "$do_pillarboxing" == "True"; then 
  ...
   ```
* The `awk` calls can be done with `$(python -c 'printf(round(\$var*1.3333))')"`. This will make it easier to understand the intentions, since there's no need for workarounds to get a rounded value (i.e. add `+0.5` and count on `printf` to truncate down).

Reply:
Did not consider python in the first place but right, it removes a dependency and is less awk-ward, changed with pleasure. Used python3 explicitly as with python2 there would have been additional float/int casts needed.

Reply:
Yeah, I think it's simpler now. Thanks for the changes.